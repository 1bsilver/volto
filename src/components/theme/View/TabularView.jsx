/**
 * Tabular view component.
 * @module components/theme/View/TabularView
 */

import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { Link } from 'react-router';
import { Table } from 'semantic-ui-react';

/**
 * Tabular view component class.
 * @function TabularView
 * @params {object} content Content object.
 * @returns {string} Markup of the component.
 */
const TabularView = ({ content }) => (
  <div id="page-home">
    <Helmet title={content.title} />
    <article id="content">
      <header>
        <h1 className="documentFirstHeading">{content.title}</h1>
        {content.description &&
          <p className="description">{content.description}</p>}
      </header>
      <section id="content-core">
        <Table celled padded>
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell>Title</Table.HeaderCell>
              <Table.HeaderCell>Description</Table.HeaderCell>
              <Table.HeaderCell>Type</Table.HeaderCell>
              <Table.HeaderCell>State</Table.HeaderCell>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            {content.items.map(item => (
              <Table.Row key={item.url}>
                <Table.Cell>
                  <Link
                    to={item.url}
                    className="summary url"
                    title={item['@type']}
                  >
                    {item.title}
                  </Link>
                </Table.Cell>
                <Table.Cell>{item.description}</Table.Cell>
                <Table.Cell>{item['@type']}</Table.Cell>
                <Table.Cell>{item.review_state}</Table.Cell>
              </Table.Row>
            ))}
          </Table.Body>
        </Table>
      </section>
    </article>
  </div>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
TabularView.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string,
        description: PropTypes.string,
        url: PropTypes.string,
        review_state: PropTypes.string,
        '@type': PropTypes.string,
      }),
    ),
  }).isRequired,
};

export default TabularView;
