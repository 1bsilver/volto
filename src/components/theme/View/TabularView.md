TabularView example source:

```jsx_ noeditor
const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;
import TabularView from './TabularView';
import {StaticRouter} from 'react-router-dom';
const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
  users: {
    reset: {
      loading: '',
    },
  },
  actions: {},
  content: { get: {} },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <TabularView
        content={{
          title: 'plone',
          description: 'open source CMS',
          items: [
            {
              '@id': '@plone',
              '@type': '@image',
              url: 'www.plone.org',
              title: 'plone',
              description: 'open source CMS',
              image: { scales: { thumb: {} } },
              review_state: '',
            },
          ],
        }}
      />
    </StaticRouter>
  </Provider>
</div>;
```
