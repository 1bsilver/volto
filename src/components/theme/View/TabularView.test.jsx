import React from 'react';
import renderer from 'react-test-renderer';
import TabularView from './TabularView';

test('renders a tabular view component', () => {
  const component = renderer.create(
    <TabularView
      content={{
        title: 'Hello World!',
        description: 'Hi',
        items: [
          {
            title: 'My item',
            description: 'My item description',
            url: 'http://item',
          },
        ],
      }}
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
