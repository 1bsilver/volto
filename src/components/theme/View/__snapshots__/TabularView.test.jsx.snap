// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a tabular view component 1`] = `
<div
  id="page-home"
>
  <article
    id="content"
  >
    <header>
      <h1
        className="documentFirstHeading"
      >
        Hello World!
      </h1>
      <p
        className="description"
      >
        Hi
      </p>
    </header>
    <section
      id="content-core"
    >
      <table
        className="ui celled padded table"
      >
        <thead
          className=""
        >
          <tr
            className=""
          >
            <th
              className=""
            >
              Title
            </th>
            <th
              className=""
            >
              Description
            </th>
            <th
              className=""
            >
              Type
            </th>
            <th
              className=""
            >
              State
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        >
          <tr
            className=""
          >
            <td
              className=""
            >
              <a
                className="summary url"
                onClick={[Function]}
                style={Object {}}
                title={undefined}
              >
                My item
              </a>
            </td>
            <td
              className=""
            >
              My item description
            </td>
            <td
              className=""
            />
            <td
              className=""
            />
          </tr>
        </tbody>
      </table>
    </section>
  </article>
</div>
`;
